module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.cpp

```diff
@@ -77,64 +120,166 @@ namespace behaviour {
                     case KEY_UP: lookUp(); break;
                     case KEY_DOWN: lookDown(); break;
                     case 'q': quit(); return;
-                    default: log("Unknown Command");
+                    default:
+                        log<NUClear::ERROR>("Unknown Command");
+                        printStatus();
+                        break;
+                }
+            });
+
+            on<Trigger<LogMessage>, Sync<KeyboardWalk>>().then([this](const LogMessage& packet) {
+                // Where this message came from
+                std::string source = "";
+
+                // If we know where this log message came from, we display that
+                if (packet.task) {
+                    // Get our reactor name
+                    std::string reactor = packet.task->identifier[1];
```

Comment:
Currently this line causes a set fault. Not sure why.

Reply:
Not every log line has a reactor name... not sure if that would cause this though

Reply:
It seems like the `task` pointer is being deallocated i between the if statement and `std::string reactor = packet.task->identifier[1];` 

Either that or its already an invalid pointer by the time we reach the if statement